src/offline.cpp

```diff
@@ -23,10 +23,50 @@ Q_DECLARE_LOGGING_CATEGORY(PACKAGEKITQT_OFFLINE)
 
 using namespace PackageKit;
 
+Offline::Results::Results(const QDBusPendingCall &call)
+    : m_reply(call)
+{
+    m_reply.waitForFinished();
+}
+
+bool Offline::Results::isError() const
+{
+    return m_reply.isError();
+}
+
+bool Offline::Results::success() const
+{
+    return m_reply.argumentAt<0>();
+}
+
+QStringList Offline::Results::packageIds() const
+{
+    return m_reply.argumentAt<1>();
+}
+
+Transaction::Role Offline::Results::role() const
+{
+    return static_cast<Transaction::Role>(m_reply.argumentAt<2>());
+}
+
+qulonglong Offline::Results::timeFinished() const
+{
+    return m_reply.argumentAt<3>();
+}
+
+Transaction::Error Offline::Results::error() const
+{
+    return static_cast<Transaction::Error>(m_reply.argumentAt<4>());
+}
+
+QString Offline::Results::errorDescription() const
+{
+    return m_reply.argumentAt<5>();
+}
+
 Offline::Offline(QObject *parent) : QObject(parent)
   , d_ptr(new OfflinePrivate(this))
 {
-    qRegisterMetaType<Transaction::Role>();
```

Comment:
Does it also fail for you with this? Or is it for general cleanliness?

Reply:
This line alone does not fix the problem for me, it still says "the type is not registered with DBus". In my approach this line is useless, so I removed it.